pkg/resource/deploy/plan_executor.go

```diff
@@ -189,13 +193,22 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 	pe.stepExec.WaitForCompletion()
 	logging.V(4).Infof("planExecutor.Execute(...): step executor has completed")
 
+	if res != nil && res.IsBail() {
+		return res
+	}
+
+	var err error
+	if res != nil {
+		err = res.Error()
+	}
+
 	// Figure out if execution failed and why. Step generation and execution errors trump cancellation.
 	if err != nil || pe.stepExec.Errored() {
 		err = execError("failed", preview)
 	} else if canceled {
 		err = execError("canceled", preview)
```

Comment:
I'm thinking this should be a bail as well; cancellation doesn't sound to me like something that should be a non-bail Result.